rgw/v2/tests/s3cmd/test_rate_limit.py

```diff
@@ -0,0 +1,313 @@
+        )
+        stdout, stderr = rc.communicate()
+        log.info(stdout)
+        log.info(stderr)
+    except Exception as e:
+        raise S3CommandExecError(message=str(e))
+    assert "503" in str(stderr), "Rate limit slowdown not observed, failing!"
+
+    # sleep a minute to reset the ops limit
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+
+    # test the read and write data limit
+    try:
+        range_val = "{1..3}"
+        cmd = (
+            f"for i in {range_val}; do /home/cephuser/venv/bin/s3cmd ls "
+            f"s3://{bucket_name2}/files/file1 ;done;"
+        )
+        # rc = utils.exec_shell_cmd(cmd)
+        rc = subprocess.Popen(
+            [cmd], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        stdout, stderr = rc.communicate()
+        log.info(stdout)
+        log.info(stderr)
+    except Exception as e:
+        raise S3CommandExecError(message=str(e))
+    assert "503" in str(stderr), "Rate limit slowdown not observed, failing!"
+
+    log.info(f"Sleeping for a minute to reset limits")
+    sleep(61)
+    try:
+        range_val = "{1..3}"
+        cmd = (
+            f"for i in {range_val}; do /home/cephuser/venv/bin/s3cmd "
+            f"put file1 s3://{bucket_name2}/files/file$i ;done;"
+        )
+        # rc = utils.exec_shell_cmd(cmd)
+        rc = subprocess.Popen(
+            [cmd], shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE
+        )
+        stdout, stderr = rc.communicate()
+        log.info(stdout)
+        log.info(stderr)
+    except Exception as e:
+        raise S3CommandExecError(message=str(e))
+    assert "503" in str(stderr), "Rate limit slowdown not observed, failing!"
+
+
+if __name__ == "__main__":
+
+    test_info = AddTestInfo("test bucket and user rate limits")
+
+    try:
+        project_dir = os.path.abspath(os.path.join(__file__, "../../.."))
+        test_data_dir = "test_data"
+        TEST_DATA_PATH = os.path.join(project_dir, test_data_dir)
+        log.info(f"TEST_DATA_PATH: {TEST_DATA_PATH}")
+        if not os.path.exists(TEST_DATA_PATH):
+            log.info("test data dir not exists, creating.. ")
+            os.makedirs(TEST_DATA_PATH)
+        parser = argparse.ArgumentParser(
+            description="RGW S3 bucket and user rate limits"
+        )
+        parser.add_argument(
+            "-c", dest="config", help="RGW S3 bucket and user rate limits"
+        )
+        parser.add_argument(
+            "-log_level",
+            dest="log_level",
+            help="Set Log Level [DEBUG, INFO, WARNING, ERROR, CRITICAL]",
+            default="info",
+        )
+        args = parser.parse_args()
+        yaml_file = args.config
+        log_f_name = os.path.basename(os.path.splitext(yaml_file)[0])
+        configure_logging(f_name=log_f_name, set_level=args.log_level.upper())
+        config = resource_op.Config(yaml_file)
+        config.read()
+        test_exec(config)
```

Comment:
Please add support of remote execution from client node

Reply:
@TejasC88 As discussed in our call, I am closing this comment since s3cmd script will run from any node without mentioning remote RGW node